sycl/test-e2e/AddressSanitizer/invalid-argument/bad-context.cpp

```diff
@@ -1,4 +1,4 @@
-// REQUIRES: linux, cpu && (gpu && level_zero)
+// REQUIRES: linux, cpu || (gpu && level_zero)
```

Comment:
According to other tests in this dir - I believe it's just a typo.

Reply:
No, this is not a typo, but maybe I used wrong pattern.
I want this test can be tested on machines which have both CPU and GPU, as you can see
```
sycl::queue gpu_queue(sycl::gpu_selector_v);
sycl::queue cpu_queue(sycl::cpu_selector_v);
```

Reply:
Okay, I got it, I'll update the REQUIRES string

Reply:
I've updated this string - [7163eb7](https://github.com/intel/llvm/pull/18007/commits/7163eb7dde6d61b311ccc79e7179a3fe44f8a144)
Anyways, looks like this test wasn't launched in [pre-commit on gen12](https://github.com/intel/llvm/actions/runs/14879362060/job/41784965360?pr=18007) - it's in the list of unsupported tests.
@intel/dpcpp-devops-reviewers could you help me to figure out if it's true?

Reply:
yep, definitely looks like it didn't run. i dont think `cpu` and `gpu` will ever both be true, i think we need `any-device-is-cpu && any-device-is-gpu`

Reply:
oh sorry, i thought the above code was the latest, let me test it

Reply:
Ah ok, the problem is that `level_zero` is only true when the device is `level_zero:gpu`, so when running on `opencl:cpu` it's not set, so the condition is never true. any `any-device-is` stuff is working as expected

i think the simplest fix is just to make it `any-device-is-cpu && (gpu && level_zero)`

Reply:
OK, I'll try

Reply:
@sarnex take a look at this test - https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/AddressSanitizer/aot/cpu.cpp
the REQUIRES there is even simpler and it also uses run-aux, but it's also in the list of unsupported tests

Reply:
ah ok yeah good find, let me actually debug it, thanks

Reply:
lol found the cause 
https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/AddressSanitizer/lit.local.cfg#L16

Reply:
Thank you for correcting the require pattern. 
Yep, we haven't fixed the GEN12 issue yet, because it's low priority.
We will take a look to GEN12 in the future.